model_compression_toolkit/common/fusion/layer_fusing.py

```diff
@@ -0,0 +1,105 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import copy
+from typing import Any, List
+from model_compression_toolkit.common.graph.base_graph import Graph
+from model_compression_toolkit.common.graph.base_node import BaseNode
+from model_compression_toolkit.common.target_platform.targetplatform2framework import TargetPlatformCapabilities
+from model_compression_toolkit.common.target_platform.targetplatform2framework.layer_filter_params import LayerFilterParams
+
+def filter_fusing_patterns(fusing_patterns: List[List[Any]], node: BaseNode, idx: int = 0):
+    """
+    Update relevant fusing patterns object if layer number 'idx' inside the fusion matches the node
+    Args:
+        fusing_patterns: supported fusings
+        node: node to decide if it can be a part of fusion
+        idx: index of layer in the fusion
+    Returns:
+        fusing_patterns after filtering non-relevant fusions
+    """
+    valid_fusing_patterns = []
+    for i,fusing_pattern in enumerate(fusing_patterns):
+        if idx < len(fusing_pattern):
+            if (type(fusing_pattern[idx]) == LayerFilterParams and fusing_pattern[idx].match(node)) or fusing_pattern[idx] == node.type:
+                valid_fusing_patterns.append(fusing_pattern)
+
+    # Return only valid patterns for this node
+    return valid_fusing_patterns
+
+
+def mark_fusing_activation(nodes: List[BaseNode]):
+    """
+    Mark activation for non-quantization needed due to fusion
+    Args:
+        nodes: nodes to update their activation quantization
+    """
+    for node in nodes:
+        for qc in node.candidates_quantization_cfg:
+            qc.activation_quantization_cfg.enable_activation_quantization = False
+
+
+def fusion(graph: Graph, tpc: TargetPlatformCapabilities) -> Graph:
+    """
+    There are cases (layers fusion) we do not need to quantize a layer's output since there is an
+    activation layer that follows it. Thus, in these cases we set the node's attribute
+    that indicates whether the output should be quantized to False.
+    tpc (TargetPlatformCapabilities) holds the fusing information.
+    This function searches and marks activation for no quantization if fusion is found.
+    Args:
+        graph: Graph we apply the fusion on.
+        tpc: TargetPlatformCapabilities object which holds fusion configuration
+    Returns:
+        Graph after applying fusion activation marking.
+    """
+    fusing_patterns = tpc.get_fusing_patterns()
+    if len(fusing_patterns) == 0:
+        return graph
+
+    max_layers_fusing = 0
+    for fusing_pattern in fusing_patterns:
+        num_layers_fusing = len(fusing_pattern)
+        max_layers_fusing = max(num_layers_fusing, max_layers_fusing)
+
```

Comment:
We can write it in a more pythonic way

Reply:
max_layer = 0 if len(fusing_patterns)==0 else max([len(f) for f in fusing_patterns])

Reply:
Done